torch/nested/_nestedtensor.py

```diff
@@ -0,0 +1,100 @@
+import torch
+from functools import wraps
+
+
+@wraps(torch._nested_tensor)
+def nested_tensor(*args, **kwargs):
+    return NestedTensor(torch._nested_tensor(*args, **kwargs))
+
+
+class NestedTensor:
+    # data is a torch.Tensor backed by a NestedTensorImpl
+
+    def __init__(self, impl):
+        self._impl = impl
+
+    @property
+    def dtype(self):
+        """
+        The data type of ```self``` NestedTensor.
+        """
+        return self._impl.dtype
+
+    @property
+    def layout(self):
+        """
+        The layout of ```self``` NestedTensor.
+        """
+        return self._impl.layout
+
+    @property
+    def device(self):
+        """
+        The device of ```self``` NestedTensor.
+        """
+        return self._impl.device
+
+    @property
+    def requires_grad(self):
+        """
+        Is ```True``` if gradients need to be computed for this Tensor.
+        """
+        return self._impl.requires_grad
+
+    def stride(self):
+        """
+        The stride of ```self``` NestedTensor.
+        """
+        return self._impl.stride()
+
+    def size(self):
+        """
+        The size of ```self``` NestedTensor.
+        """
+        return self._impl.size()
```

Comment:
why does this exist if it doesn't have a size?

Reply:
To throw an exception

Reply:
if that's expected to always be the case, perhaps that should be noted in the docstring rather than being a surprise?

Reply:
Sure, let me update